metrics_block.py

```diff
@@ -54,6 +65,56 @@ def stop(self):
         if self._metrics_job is not None:
             self._metrics_job.cancel()
 
+    def timestamp(self):
+        '''returns the current system timestamp'''
+        return datetime.isoformat(datetime.utcnow())
+
+    @staticmethod
```

Comment:
It's just something that popped up from my code viewer. Some PEP says that if it "could be a function" you should label it as a static method. This doesn't use any class variables, so it can be.